### DISCUSSION TITLE
Fallback to renovate.json file as a preset is deprecated

### DISCUSSION THREAD

**Discussion by maxs-rose** [renovatebot/renovate#14225] [Request Help] [answered]
When onboaring a new repo with GitLab there is this message in the logs:
```
DEBUG: Checking if this org/owner has a default Renovate preset which can be used. (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
DEBUG: GitLab API 404 (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "url": "https://gitlab.com/api/v4/projects/orchard-systems%2Frenovate-config/repository/branches"
DEBUG: Failed to retrieve default.json from repo (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "statusCode": 404,
       "url": "https://gitlab.com/api/v4/"
 INFO: Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead. (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
```
To setup the runner we have simply included the base template and are running with auto-onboarding:
```yml
include:
  - project: "renovate-bot/renovate-runner"
    file: "/templates/renovate-dind.gitlab-ci.yml"

renovate:
  scripts:
    - renovate --onboarding=true $RENOVATE_EXTRA_FLAGS
```

Why is this info being displayed in the logs and is there any way to remove it?

**Comment by viceice** [comment, marked as answer]
Do not use `renovate.json` files as shared preset.

**Reply by maxs-rose** [reply]
how do you configure that? All that is our config is a `hostRules` and then the override for auto onboarding

**Reply by viceice** [reply]
Please share your full sanitized config

**Reply by maxs-rose** [reply]
`.gitlab-ci.yml`:
```yml
include:
  - project: "renovate-bot/renovate-runner"
    file: "/templates/renovate-dind.gitlab-ci.yml"

renovate:
  scripts:
    - renovate --onboarding=true $RENOVATE_EXTRA_FLAGS
```

`config.js`:
```js
module.exports = {
  hostRules: [
    { ... },
  ],
  includeForks: true,
};
```

`RENOVATE_EXTRA_FLAGS` is just set to the repo path, the repo is just an angular project that does not contain any of the files that renovate is looking for
`RENOVATE_CONFIG_FILE` is `config.js`

The full log is:

```
DEBUG: Checking if this org/owner has a default Renovate preset which can be used. (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
DEBUG: GitLab API 404 (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "url": "https://gitlab.com/api/v4/projects/orchard-systems%2Frenovate-config/repository/branches"
DEBUG: Failed to retrieve default.json from repo (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "statusCode": 404,
       "url": "https://gitlab.com/api/v4/"
 INFO: Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead. (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
DEBUG: GitLab API 404 (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "url": "https://gitlab.com/api/v4/projects/orchard-systems%2Frenovate-config/repository/branches"
DEBUG: Failed to retrieve renovate.json from repo (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "statusCode": 404,
       "url": "https://gitlab.com/api/v4/"
DEBUG: GitLab API 404 (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "url": "https://gitlab.com/api/v4/projects/orchard-systems%2F.gitlab/repository/branches"
DEBUG: Failed to retrieve renovate-config.json from repo (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "statusCode": 404,
       "url": "https://gitlab.com/api/v4/"
DEBUG: No default org/owner preset found, so the default onboarding config will be used instead. Note: do not be concerned with any 404 messages that preceded this. (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
DEBUG: onboarding config (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
       "config": { "$schema": "https://docs.renovatebot.com/renovate-schema.json", "extends": ["config:base"] }
DEBUG: No renovate config file found (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
DEBUG: migrateAndValidate() (repository=orchard-systems/analytics/income/experimental/ia-angular-gui-renovate-test)
```

**Reply by maxs-rose** [reply]
This `https://gitlab.com/api/v4/projects/orchard-systems%2Frenovate-config/repository/branches` does not exist so its not finding a `renovate.json` file there.

**Reply by viceice** [reply]
I still see no issue there, i only see some debug and one info message. So what's the issue?

**Reply by maxs-rose** [reply]
Oh, so the INFO log line is just telling me that something I'm not using is being deprecated at some point then? 

If so it might be best to only show that when its actually being used

**Reply by viceice** [reply]
Suppression isn't easy doable, but maybe we can delay the message after we tried to fetch the deprecated file and only log if it's found.

@rarkins WDYT? Not sure if it's easily doable.

**Reply by maxs-rose** [reply]
Could it just log if that request returns a non 400 response possibly?

But otherwise ill mark this question as answered thanks for your help 🙂

**Reply by maxs-rose** [reply]
Here https://github.com/renovatebot/renovate/blob/c3ad246a998709852b1c8fdd91e5eadd2c70aea3/lib/config/presets/util.ts#L38-L46.

Instead of logging before it could it not simply be:

```js
jsonContent = await fetch(
        pkgName,
        buildFilePath('renovate.json'),
        endpoint,
        packageTag
      );

if(jsonContent.status === 200) {
       logger.info('Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead.');
}
```

**Reply by viceice** [reply]
Sure, but without your condition. 😉
The result of fetch is already the JSON object.

**Reply by rarkins** [reply]
I think it still needs an `if`. Won't it be `null` if not found?

**Reply by viceice** [reply]
Throws if not found

**Reply by maxs-rose** [reply]
If you guys like the idea of just moving the log line down I don't mind making a pull request for it

**Reply by viceice** [reply]
Please do. 🤗

**Reply by maxs-rose** [reply]
Created #14281

**Comment by razzeee** [comment]
I guess it's caused by the docker image having set
`RENOVATE_ONBOARDING_CONFIG: '{"$$schema": "https://docs.renovatebot.com/renovate-schema.json", "extends": ["config:base"] }'`

and https://docs.renovatebot.com/renovate-schema.json configures this

```json
  "onboardingConfigFileName": {
      "description": "Change this value to override the default onboarding config file name.",
      "type": "string",
      "default": "renovate.json"
    },
```

**Reply by viceice** [reply]
No, this alone isn't causing that fallback message

**Reply by razzeee** [reply]
So that default doesn't lead to the onboarding creating `renovate.json` files? Where would that happen then, as I haven't set it in any way.

**Reply by viceice** [reply]
sure, onboarding will create that file, but renovate shouldn't onboard a repo for shared renovate presets. and you shouldn't use that repo for other stuff.

If you have a renovate.json in that repo, rename it to any of the other allowed file names.

**Reply by razzeee** [reply]
not sure what shared renovate presets are, but onboarding did create `renovate.json` and then directly started to complain in the log about it, so that is kinda uncool.

**Reply by rarkins** [reply]
Then don't use renovate-config as the repo name

**Reply by razzeee** [reply]
Interesting, how is that a factor?

**Reply by viceice** [reply]
Yes, renovate checks that repo for a default organization onboarding config `<org>/renovate-config/default.json` (fallback `<org>/renovate-config/renovate.json`).

It also checks `<org>/.github/default.json` (fallback `<org>/.github/renovate.json`).

**Reply by razzeee** [reply]
Thank you, renamed the repo from `renovate-config` to something else and it's not complaining anymore.
